openlibrary/catalog/add_book/__init__.py

```diff
@@ -969,6 +1020,11 @@ def load(rec, account_key=None):
         work = new_work(edition.dict(), rec)
         edition.works = [{'key': work['key']}]
 
+    if should_overwrite_promise_item(
+        edition=edition, from_marc_record=from_marc_record
+    ):
+        return load_data(rec, account_key=account_key, existing_edition=edition)
```

Comment:
Good place for a comment explaining why we're treating this almost like a new record.

Reply:
I added an explanatory comment.